{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.1",
    "cveMetadata": {
        "cveId": "CVE-2024-4418",
        "assignerOrgId": "53f830b8-0a3f-465b-8143-3b8a9948e749",
        "state": "PUBLISHED",
        "assignerShortName": "redhat",
        "dateReserved": "2024-05-02T10:52:32.129Z",
        "datePublished": "2024-05-08T03:03:05.135Z",
        "dateUpdated": "2024-06-21T16:23:11.284Z"
    },
    "containers": {
        "cna": {
            "title": "Libvirt: stack use-after-free in virnetclientioeventloop()",
            "metrics": [
                {
                    "other": {
                        "content": {
                            "value": "Moderate",
                            "namespace": "https://access.redhat.com/security/updates/classification/"
                        },
                        "type": "Red Hat severity rating"
                    }
                },
                {
                    "cvssV3_1": {
                        "attackComplexity": "LOW",
                        "attackVector": "LOCAL",
                        "availabilityImpact": "HIGH",
                        "baseScore": 6.2,
                        "baseSeverity": "MEDIUM",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                        "version": "3.1"
                    },
                    "format": "CVSS"
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "A race condition leading to a stack use-after-free flaw was found in libvirt. Due to a bad assumption in the virNetClientIOEventLoop() method, the `data` pointer to a stack-allocated virNetClientIOEventData structure ended up being used in the virNetClientIOEventFD callback while the data pointer's stack frame was concurrently being \"freed\" when returning from virNetClientIOEventLoop(). The 'virtproxyd' daemon can be used to trigger requests. If libvirt is configured with fine-grained access control, this issue, in theory, allows a user to escape their otherwise limited access. This flaw allows a local, unprivileged user to access virtproxyd without authenticating. Remote users would need to authenticate before they could access it."
                }
            ],
            "affected": [
                {
                    "vendor": "Red Hat",
                    "product": "Red Hat Enterprise Linux 6",
                    "collectionURL": "https://access.redhat.com/downloads/content/package-browser/",
                    "packageName": "libvirt",
                    "defaultStatus": "unknown",
                    "cpes": [
                        "cpe:/o:redhat:enterprise_linux:6"
                    ]
                },
                {
                    "vendor": "Red Hat",
                    "product": "Red Hat Enterprise Linux 7",
                    "collectionURL": "https://access.redhat.com/downloads/content/package-browser/",
                    "packageName": "libvirt",
                    "defaultStatus": "unknown",
                    "cpes": [
                        "cpe:/o:redhat:enterprise_linux:7"
                    ]
                },
                {
                    "vendor": "Red Hat",
                    "product": "Red Hat Enterprise Linux 8",
                    "collectionURL": "https://access.redhat.com/downloads/content/package-browser/",
                    "packageName": "virt:rhel/libvirt",
                    "defaultStatus": "affected",
                    "cpes": [
                        "cpe:/o:redhat:enterprise_linux:8"
                    ]
                },
                {
                    "vendor": "Red Hat",
                    "product": "Red Hat Enterprise Linux 8 Advanced Virtualization",
                    "collectionURL": "https://access.redhat.com/downloads/content/package-browser/",
                    "packageName": "virt:av/libvirt",
                    "defaultStatus": "affected",
                    "cpes": [
                        "cpe:/a:redhat:advanced_virtualization:8::el8"
                    ]
                },
                {
                    "vendor": "Red Hat",
                    "product": "Red Hat Enterprise Linux 9",
                    "collectionURL": "https://access.redhat.com/downloads/content/package-browser/",
                    "packageName": "libvirt",
                    "defaultStatus": "affected",
                    "cpes": [
                        "cpe:/o:redhat:enterprise_linux:9"
                    ]
                }
            ],
            "references": [
                {
                    "url": "https://access.redhat.com/security/cve/CVE-2024-4418",
                    "tags": [
                        "vdb-entry",
                        "x_refsource_REDHAT"
                    ]
                },
                {
                    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2278616",
                    "name": "RHBZ#2278616",
                    "tags": [
                        "issue-tracking",
                        "x_refsource_REDHAT"
                    ]
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4IE44UIIC3QWBFRB4EUSFNLJBU6JLNSD/"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/Q4ZQBAJVHIZMCZNTRPUW3ZKXRKLXRQZU/"
                }
            ],
            "datePublic": "2024-05-02T00:00:00+00:00",
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-416",
                            "description": "Use After Free",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "x_redhatCweChain": "CWE-362->CWE-416: Concurrent Execution using Shared Resource with Improper Synchronization ('Race Condition') leads to Use After Free",
            "timeline": [
                {
                    "lang": "en",
                    "time": "2024-05-02T00:00:00+00:00",
                    "value": "Reported to Red Hat."
                },
                {
                    "lang": "en",
                    "time": "2024-05-02T00:00:00+00:00",
                    "value": "Made public."
                }
            ],
            "credits": [
                {
                    "lang": "en",
                    "value": "Red Hat would like to thank Martin Å irokov for reporting this issue."
                }
            ],
            "providerMetadata": {
                "orgId": "53f830b8-0a3f-465b-8143-3b8a9948e749",
                "shortName": "redhat",
                "dateUpdated": "2024-05-08T03:03:05.135Z"
            }
        },
        "adp": [
            {
                "providerMetadata": {
                    "orgId": "96f1cf3c-f80a-4e0f-a063-99a91e3f1325",
                    "shortName": "secretariat-reference",
                    "dateUpdated": "2024-06-21T16:23:11.284Z"
                },
                "references": [
                    {
                        "url": "https://access.redhat.com/security/cve/CVE-2024-4418/DEMO_2024_06_19",
                        "tags": [
                            "vdb-entry",
                            "x_refsource_REDHAT",
                            "x_cve_prog_secretariat_snapshot"
                        ]
                    },
                    {
                        "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2278616/DEMO_2024_06_19",
                        "name": "RHBZ#2278616",
                        "tags": [
                            "issue-tracking",
                            "x_refsource_REDHAT",
                            "x_cve_prog_secretariat_snapshot"
                        ]
                    }
                ]
            }
        ]
    }
}